### ISSUE TITLE
Dependency updates for Prometheus Federator, helm project operator and helm locker

### ISSUE THREAD

**Issue by MKlimuszka** [rancher/rancher#46296]
We want to work towards a versioning scheme of dependencies in :

- https://github.com/rancher/helm-locker
- https://github.com/rancher/helm-project-operator
- https://github.com/rancher/prometheus-federator

That will be guaranteed to be compatible with specific versions of k8s / rancher and be maintainable through several releases,  as well as having e2e tests to verify behaviour of these controllers.

**Comment by alexandreLamarre** [comment]
My initial proposal is to consolidate all these controller implementations into a single repository, likely https://github.com/rancher/prometheus-federator, so that there is a single point to manage dependencies in a consistent way across the controller implementations. This way we can have one repository with specific release branches, following the semcov of something like [BRO](https://github.com/rancher/backup-restore-operator/),  with release/v5.0 branch and images 0.5.X being compatible with the k8s versions in rancher 2.9


To achieve that, I propose to :
- Write some comprehensive integration/e2e tests for these controllers behaviour, like in https://github.com/rancher/helm-locker/pull/21
- Once that is done, aggressively bump dependencies in each of these repos to ones compatible with rancher 2.9
- Cut an rc for prometheus federator with the dependency bumps and pass that through QA
- Combine the controller implementations into prometheus-federator

**Comment by mallardduck** [comment]
I have created a [new PR](https://github.com/rancher/helm-project-operator/pull/74) that merges the `helm-locker` repo with the `helm-project-operator` repo. This was done in a manner that preserves ability to build and test `helm-locker` alone. However will reduce our current workflow burden for updating things in an effective manner.

Needs review by @joshmeranda when he is back, and then some follow up CI work based on his input. After merging that PR we can then "port" (lack of better term) the version bumps and improvements (that apply) from [this PR](https://github.com/rancher/helm-locker/pull/24) alex had started into this repo.

We will also need to implement the proper branching policy per the ADR - we should decide if that should be done before or after porting PR 24 alex started over to this repo. As if those changes may affected older branches we need to create, we should create those first to preserve compatibility.

**Comment by mallardduck** [comment]
I have created a set of PRs, images and chart changes to allow for QA to test before merge. An issue to track that specific process is here: https://github.com/rancher/prometheus-federator/issues/94

I figured it makes sense for QA to use that issue instead of this one - since this one is really kind of an "Epic". And we're breaking it into multiple phases to make the huge effort more digestable.

**Comment by mallardduck** [comment]
I've completed a knowledge transfer on this effort to @jbiers - so she has some foundational understanding and can potentially work with Josh on this a bit when he's back (time permitting). The changes in my PRs will need to be merged first, but after that I also have a branch prepared to address https://github.com/rancher/rancher/issues/46764

**Comment by mallardduck** [comment]
Hit another milestone today - we merged the PR which combines both HPO+HL into PromFed. Working towards tagging a new PromFed RC version and updating charts.

**Comment by mallardduck** [comment]
Since we've completed merging code-bases of the projects and it's been QA'd once. I've created the PR for charts updates here: https://github.com/rancher/charts/pull/4588

**Comment by mallardduck** [comment]
Added @alexandreLamarre as an Assignee as he'll be working on the PRs to update dependencies of PromFed so he'll be closing out the final phase of the project which should solve this issue.

**Comment by jbiers** [comment]
@alexandreLamarre do we plan on backporting the dependency updates to v2.9.6?

**Comment by alexandreLamarre** [comment]
Yes, with the matching dependency versions for the supported k8s distribution

**Comment by alexandreLamarre** [comment]
@pankajsqe hit a helm template error while installing the chart for rancher 2.11, I've verified on rancher 2.11 that installing from https://github.com/rancher/prometheus-federator/releases/download/v3.0.0-rc.2/prometheus-federator-3.0.0-rc.2.tgz succeeds so I think that narrows down the bug to the way `rancher-charts` is serving `106.0.1-rc.1+up3.0.0-rc.2`

**Comment by alexandreLamarre** [comment]
The way the chart was prepared for that particular version still relied on `helmProjectOperator` subchart as a dependency, although that is invalid with the current way the new prometheus federator chart works. Creating a new build now without that dependency.

**Comment by alexandreLamarre** [comment]
Verfied the new build   `106.0.1-rc.3+up3.0.0-rc.2` install successfully when service from rancher/charts repo:
```
Release "prometheus-federator" has been upgraded. Happy Helming!
NAME: prometheus-federator
LAST DEPLOYED: Thu Feb 20 11:12:15 2025
NAMESPACE: cattle-monitoring-system
STATUS: deployed
REVISION: 3
TEST SUITE: None
NOTES:
prometheus-federator has been installed. Check its status by running:
  kubectl --namespace cattle-monitoring-system get pods -l "release=prometheus-federator"

```

**Comment by pankajsqe** [comment]
Prometheus federator chart 106.0.1-rc.3+up3.0.0-rc.2 is working as expected.

details
rancher : v2.11.0-alpha5
cluster : v1.31.1+rke2r1
prom-fed : 106.0.1-rc.3+up3.0.0-rc.2



<img width="1726" alt="Image" src="https://github.com/user-attachments/assets/ed0258bb-67a4-4a5e-807b-f521424a9ca9" />
<img width="1728" alt="Image" src="https://github.com/user-attachments/assets/79f35f78-12e7-4d31-9a93-95ce47cc6e97" />
<img width="1727" alt="Image" src="https://github.com/user-attachments/assets/0d49b3ad-20ed-4fd3-94f9-557b4d78fd36" />
